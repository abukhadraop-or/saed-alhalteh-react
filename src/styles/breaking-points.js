const breakingPoints = {
  mobile: '30rem',
  tablets: '48rem',
  smallScreen: '64.0625rem',
  largeScreen: '75rem',
};

export default breakingPoints;
