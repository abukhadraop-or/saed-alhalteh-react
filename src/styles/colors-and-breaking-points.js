export const colors = {
  black: '#000',
  blueViolet: '#4F52B0',
  concrete: '#f3f3f3',
  dustyGrey: '#999',
  edward: '#acb9b4',
  fern: '#5cb85c',
  goblin: '#347634',
  gray: '#868686',
  mineShaft: '#333333',
  oBlack: 'rgba(0,0,0,0.75)',
  o3Black: 'rgba(0,0,0,0.3)',
  oFern: 'rgba(92,184,92,0.5)',
  silver: '#c8c8c8',
  white: '#fff',
};

export const breakingPoints = {
  mobile: '480px',
  tablets: '768px',
  smallScreen: '1025px',
  largeScreen: '1200px',
};
