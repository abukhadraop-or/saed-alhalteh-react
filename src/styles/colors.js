const colors = {
  black: '#000000',
  blueViolet: '#4f52b0',
  concrete: '#f3f3f3',
  darkGreen: '#347634',
  dustyGrey: '#999999',
  gray: '#868686',
  green: '#5cb85c',
  lightGrey: '#acb9b4',
  mineShaft: '#333333',
  partiallyDarkBlack: 'rgba(0,0,0,0.75)',
  silver: '#c8c8c8',
  transparentBlack: 'rgba(0,0,0,0.3)',
  transparentGreen: 'rgba(92,184,92,0.5)',
  white: '#ffffff',
};

export default colors;
