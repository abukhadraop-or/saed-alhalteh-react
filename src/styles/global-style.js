import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body {
    font-family:'Titillium Web', sans-serif;
    margin: 0;
  }
`;

export default GlobalStyle;
