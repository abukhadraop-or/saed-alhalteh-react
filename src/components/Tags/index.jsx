import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import getTags from 'services/fake-tag-lists';
import { Tag, TagsBox, TagsContainer, Title } from './tags.styles';

/**
 * Tags section.
 *
 * @param {Object} props The props object.
 * @param {Function} props.onClick Handles click events on tags.
 * @param {String[]} props.selectedTags The selected tags.
 *
 * @return {JSX.Element} Tags component.
 */
function Tags({ onClick, selectedTags }) {
  const [tags, setTags] = useState([]);

  useEffect(() => {
    const result = getTags();
    setTags(result.tags);
  }, []);

  return (
    <TagsContainer>
      <Title> Popular Tags</Title>
      <TagsBox>
        {tags.length === 0 ? (
          <b> No Tags are available yet... </b>
        ) : (
          tags.map((tag) => (
            <Tag key={tag} onClick={() => onClick(tag)} isClicked={selectedTags.includes(tag)}>
              {tag}
            </Tag>
          ))
        )}
      </TagsBox>
    </TagsContainer>
  );
}

Tags.propTypes = {
  onClick: PropTypes.func.isRequired,
  selectedTags: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Tags;
