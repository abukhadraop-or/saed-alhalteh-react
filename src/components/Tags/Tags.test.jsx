import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import getTags from 'services/fake-tag-lists';
import Tags from '.';
import { Tag } from './tags.styles';

jest.mock('services/fake-tag-lists');

const mockData = {
  tags: ['odd', 'even', 'only the first Article', '3/9/16/27'],
};
const mockOnClick = jest.fn();

let mountWrapper;
const tag = () => mountWrapper.find(Tag);

describe('<Tags /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Tags selectedTags={[]} onClick={mockOnClick} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('sets the tags in the state after fetching them from the API', () => {
    getTags.mockReturnValue(mockData);
    mountWrapper = mount(<Tags selectedTags={[]} onClick={mockOnClick} />);

    expect(tag().length).toBe(4);
  });

  it('calls onClick function when clicked', () => {
    getTags.mockReturnValue(mockData);
    mountWrapper = mount(<Tags selectedTags={[]} onClick={mockOnClick} />);

    tag().at(0).simulate('click');

    expect(mockOnClick).toHaveBeenCalledWith('odd');
  });

  it('assigns isClicked props correctly', () => {
    getTags.mockReturnValue(mockData);
    mountWrapper = mount(<Tags selectedTags={['odd']} onClick={mockOnClick} />);
    const { isClicked } = tag().at(0).props();

    expect(isClicked).toBeTruthy();
  });
});
