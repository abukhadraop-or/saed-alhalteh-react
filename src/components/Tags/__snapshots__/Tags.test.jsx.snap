// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> tests matches the snapshot 1`] = `
<styled.div>
  <styled.h2>
     Popular Tags
  </styled.h2>
  <styled.div>
    <b>
       No Tags are available yet... 
    </b>
  </styled.div>
</styled.div>
`;
